src/OrchardCore.Modules/OrchardCore.Https/Startup.cs

```diff
@@ -16,24 +14,38 @@ namespace OrchardCore.Https
 {
     public class Startup : StartupBase
     {
-        public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
+        public override void ConfigureServices(IServiceCollection services)
         {
-            var rewriteOptions = new RewriteOptions();
+            services.AddScoped<INavigationProvider, AdminMenu>();
+            services.AddScoped<IDisplayDriver<ISite>, HttpsSettingsDisplayDriver>();
+            services.AddSingleton<IHttpsService, HttpsService>();
 
-            // Configure the rewrite options.
-            serviceProvider.GetService<IConfigureOptions<RewriteOptions>>().Configure(rewriteOptions);
+            services.AddHttpsRedirection(options => {});
```

Comment:
options => {} may look odd since we don't configure any options but I want to still call this in case in the future the library registers required services in this method.